Trakt/Api/TraktApi.cs

```diff
@@ -1108,27 +1108,28 @@ private Task<Stream> PostToTrakt(string url, object data, TraktUser traktUser)
             }
         }
 
-        private async Task<T> Retry<T>(Func<Task<T>> function)
+        private async Task<HttpResponseMessage> Retry(Func<Task<HttpResponseMessage>> function)
```

Comment:
IMHO if you change the function to be non generic you could rename it to `RetryHttpRequest` or smth.